internal/repository/employee/employee.go

```diff
@@ -2,91 +2,108 @@ package employee
 
 import (
 	"context"
+	"database/sql"
 	"errors"
-	"fmt"
-	"sync"
-
 	"github.com/dilyara4949/employees-api/internal/domain"
-
-	"github.com/google/uuid"
 )
 
 type PositionsRepository interface {
 	Get(ctx context.Context, id string) (*domain.Position, error)
 }
 
 type employeeRepository struct {
-	mu            sync.RWMutex
-	storage       map[string]domain.Employee
+	db            *sql.DB
 	positionsRepo PositionsRepository
 }
 
-func NewEmployeesRepository(positionsRepo PositionsRepository) domain.EmployeesRepository {
+func NewEmployeesRepository(db *sql.DB, positionsRepo PositionsRepository) domain.EmployeesRepository {
 	return &employeeRepository{
-		storage:       make(map[string]domain.Employee),
+		db:            db,
 		positionsRepo: positionsRepo,
 	}
 }
 
+var (
+	ErrEmployeeNotFound = errors.New("employee not found")
+	ErrNothingChanged   = errors.New("nothing changed")
+)
+
 func (e *employeeRepository) Create(ctx context.Context, employee *domain.Employee) error {
 	if _, err := e.positionsRepo.Get(ctx, employee.PositionID); err != nil {
-		return fmt.Errorf("error to create employee: %w", err)
+		return err
 	}
 
-	employee.ID = uuid.New().String()
-
-	e.mu.Lock()
-	defer e.mu.Unlock()
-	e.storage[employee.ID] = *employee
+	stmt := "insert into employees (id, first_name, last_name, position_id, created_at) values ($1, $2, $3, $4, CURRENT_TIMESTAMP);"
 
+	if _, err := e.db.Exec(stmt, employee.ID, employee.FirstName, employee.LastName, employee.PositionID); err != nil {
+		return err
+	}
 	return nil
 }
 
 func (e *employeeRepository) Get(_ context.Context, id string) (*domain.Employee, error) {
-	e.mu.RLock()
-	defer e.mu.RUnlock()
-
-	if employee, ok := e.storage[id]; ok {
+	stmt := "select first_name, last_name, position_id from employees where id = $1;"
+	row := e.db.QueryRow(stmt, id)
+
+	employee := domain.Employee{}
+	switch err := row.Scan(&employee.FirstName, &employee.LastName, &employee.PositionID); err {
```

Comment:
That's not a good pattern for error checking. 
1. You should check for `nil` first to assure the early exit
2. `case sql.ErrNoRows` is essentially `err == sql.ErrNoRows` which might behave inconsistently(it fails on wrapped errors). It is advised to use `errors.Is(err, sql.ErrNoRows)` for such checks. overall the better approach would be one of the following variants
```go
if err != nil {
	if errors.Is(err, sql.ErrNoRows) {
            return nil, ErrPositionNotFound
        }
        return nil, err
}
```
```go
if err != nil {
	switch {
	case errors.Is(err, sql.ErrNoRows):
		return nil, ErrPositionNotFound
	default:
		return nil, err
	}
}
```

Reply:
fixed